cli/running/watchdog.go

```diff
@@ -35,10 +35,12 @@ type Watchdog struct {
 	// restartTimeout describes the timeout between
 	// restarting the Instance.
 	restartTimeout time.Duration
-	// stopped indicates whether Watchdog was stopped.
-	stopped bool
-	// stopMutex used to avoid a race condition under stopped field.
-	stopMutex sync.Mutex
+	// watchdogStateMutex used to avoid a race condition under started and shouldStop fields.
```

Comment:
In fact, mutex and statuses are related to an instance, not to a watchdog. It would be nice to reflect this in the attribute names.

Reply:
Fixed, thank you.

Reply:
> In fact, mutex and statuses are related to an instance, not to a watchdog. It would be nice to reflect this in the attribute names.

Cannot agree. `shouldStop` flag is watchdog responsibility, not an instance.
Instance is independent entity and shouldStop flag is required by watchdog to make some cleanup, which is not required by Instance.
`started` - it depends...

Reply:
I think, mutex should not be a part of Instance struct. Instance does not know anything about signal loop and multithreading. This protection is required only by watchdog, not an instance. So, this is watchdog responsibility to perform safe accesses to an instance.

Reply:
And Instance cannot know anything about what fields are accessed simultaneously: states only, or anything else. But this mutex makes unsubstantiated assumptions, which fields will be updated simultaneously somehow, somewhere.

Reply:
Discussed by voice - left in the watchdog struct.